pipeline/src/pipeline.rs

```diff
@@ -63,11 +62,11 @@ pub struct Artifacts<T: FieldElement> {
     /// An analyzed .pil file, with all dependencies imported, potentially from other files.
     analyzed_pil: Option<Analyzed<T>>,
     /// An optimized .pil file.
-    optimized_pil: Option<Rc<Analyzed<T>>>,
+    optimized_pil: Option<Arc<Analyzed<T>>>,
```

Comment:
in general because of plonky3 trait bounds, but maybe the change was not required on this struct, will check.

Reply:
Ok it is, because from `Pipeline` we create `WitnessCallback`, which is included in `PowdrCircuit`, which implements `BaseAir`, which requires `Sync`, which isn't implemented on `Rc`. So we need `Arc` everywhere.